x-pack/plugins/elastic_assistant/server/routes/anonymization_fields/bulk_actions_route.ts

```diff
@@ -208,7 +206,7 @@ export const bulkActionAnonymizationFieldsRoute = (
             // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
           } = await writer!.bulk({
             documentsToCreate: body.create?.map((f) =>
-              transformToCreateScheme(authenticatedUser, changedAt, f)
+              transformToCreateScheme(changedAt, f, authenticatedUser)
```

Comment:
I don't think we need to change the type in these functions to allow for `null` user. I bet the types tripped up because you removed the null check, but it still exists in `performChecks`. If it is ok with you, I am going to try to get this type back to `AuthenticatedUser` and not accept null. Also, I didn't like that we switched order of the arguments here and didn't switch them below in `transformToCreateScheme`.

If I'm right... I'll push a commit shortly 😄

Reply:
Ok so this ended up leading me to make a lot more changes, as I noticed `performChecks` is not used everywhere it should be. I'm going to make these changes in a follow up PR.

Reply:
Here is that PR. Let me know if you think it should be part of this one: https://github.com/elastic/kibana/pull/198392

Reply:
Committed to this PR, resolving!